File: scripts/fix_types.py
"""批量修复 register/page.tsx 中的 e.target.style 类型错误"""
import re

path = r'c:\seekname\src\app\register\page.tsx'

with open(path, 'r', encoding='utf-8') as f:
    content = f.read()

# 统计修复前
count_before = len(re.findall(r'e\.target\.style|e\.currentTarget\.style', content))
print(f"修复前: {count_before} 处")

# 1. input onFocus/onBlur: e.target.style.XXX = "value"
# 匹配 onFocus={(e) => (e.target.style.borderColor = "#E86A17")}
content = re.sub(
    r'onFocus=\{(e\) => \(e\.target\.style\.(\w+) = "([^"]+)"\)\}',
    r'onFocus={(e) => { (e.target as HTMLElement).style.\1 = "\2"; }}',
    content
)
content = re.sub(
    r'onBlur=\{(e\) => \(e\.target\.style\.(\w+) = "([^"]+)"\)\}',
    r'onBlur={(e) => { (e.target as HTMLElement).style.\1 = "\2"; }}',
    content
)

# 2. Link onMouseEnter/Leave: e.target.style.textDecoration = "underline"
content = re.sub(
    r'onMouseEnter=\{(e\) => \(e\.target\.style\.(\w+) = "([^"]+)"\)\}',
    r'onMouseEnter={(e) => (e.currentTarget as HTMLElement).style.\1 = "\2"}',
    content
)
content = re.sub(
    r'onMouseLeave=\{(e\) => \(e\.target\.style\.(\w+) = "([^"]+)"\)\}',
    r'onMouseLeave={(e) => (e.currentTarget as HTMLElement).style.\1 = "\2"}',
    content
)

# 3. button onMouseEnter/Leave 多行格式 (e.currentTarget)
content = re.sub(
    r'onMouseEnter=\{(e)\s*=>\s*\n?\s*\(e\.currentTarget\.style\.color = "#E86A17"\)\s*\n?\s*\}',
    r'onMouseEnter={(e) => (e.currentTarget as HTMLElement).style.color = "#E86A17"}',
    content
)
content = re.sub(
    r'onMouseLeave=\{(e)\s*=>\s*\n?\s*\(e\.currentTarget\.style\.color = "#B0AAA0"\)\s*\n?\s*\}',
    r'onMouseLeave={(e) => (e.currentTarget as HTMLElement).style.color = "#B0AAA0"}',
    content
)

# 统计修复后
count_after = len(re.findall(r'e\.target\.style|e\.currentTarget\.style', content))
print(f"修复后: {count_after} 处")

with open(path, 'w', encoding='utf-8') as f:
    f.write(content)

print("Done!")

File: scripts/fix_types.js
// 批量修复 register/page.tsx 中的 e.target.style 类型错误（字符串替换）
const fs = require('fs');
const path = 'src/app/register/page.tsx';

let content = fs.readFileSync(path, 'utf-8');
const before = (content.match(/e\.target\.style|e\.currentTarget\.style/g) || []).length;
console.log('Before:', before);

// 1) onFocus/onBlur - input fields (6 occurrences)
content = content.replaceAll(
    'onFocus={(e) => (e.target.style.borderColor = "#E86A17")}',
    'onFocus={(e) => { (e.target as HTMLElement).style.borderColor = "#E86A17"; }}'
);
content = content.replaceAll(
    'onBlur={(e) => (e.target.style.borderColor = "#DDD0C0")}',
    'onBlur={(e) => { (e.target as HTMLElement).style.borderColor = "#DDD0C0"; }}'
);

// 2) onMouseEnter/Leave - Link text-decoration
content = content.replaceAll(
    'onMouseEnter={(e) => (e.target.style.textDecoration = "underline")}',
    'onMouseEnter={(e) => (e.currentTarget as HTMLElement).style.textDecoration = "underline"}'
);
content = content.replaceAll(
    'onMouseLeave={(e) => (e.target.style.textDecoration = "none")}',
    'onMouseLeave={(e) => (e.currentTarget as HTMLElement).style.textDecoration = "none"}'
);

// 3) onMouseEnter/Leave - return-home Link color
content = content.replaceAll(
    'onMouseEnter={(e) => (e.target.style.color = "#E86A17")}',
    'onMouseEnter={(e) => (e.currentTarget as HTMLElement).style.color = "#E86A17"}'
);
content = content.replaceAll(
    'onMouseLeave={(e) => (e.target.style.color = "#B0AAA0")}',
    'onMouseLeave={(e) => (e.currentTarget as HTMLElement).style.color = "#B0AAA0"}'
);

// 4) button multi-line currentTarget
content = content.replaceAll(
    `onMouseEnter={(e) =>
                    (e.currentTarget.style.color = "#E86A17")
                  }
                  onMouseLeave={(e) =>
                    (e.currentTarget.style.color = "#B0AAA0")
                  }`,
    `onMouseEnter={(e) => (e.currentTarget as HTMLElement).style.color = "#E86A17"}
                  onMouseLeave={(e) => (e.currentTarget as HTMLElement).style.color = "#B0AAA0"}`
);

const after = (content.match(/e\.target\.style|e\.currentTarget\.style/g) || []).length;
console.log('After:', after);
fs.writeFileSync(path, content, 'utf-8');
console.log('Done!');
